tests/integration_tests/functional/test_snapshot_editor.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for snapshot-editor tool."""
+
+import platform
+
+import pytest
+
+import host_tools.cargo_build as host
+from framework import utils
+
+PLATFORM = platform.machine()
+MIDR_EL1 = hex(0x603000000013C000)
+
+
+@pytest.mark.skipif(
+    PLATFORM != "aarch64",
+    reason="This is aarch64 specific test.",
+)
+def test_remove_regs(uvm_nano, microvm_factory):
+    """
+    This test verifies `remove-regs` method of `snapshot-editor`.
+    Here we create snapshot and try to romeve MIDR_EL1 register
+    from it. Then we try to restore uVM from the snapshot.
+    """
+
+    vm = uvm_nano
+    vm.add_net_iface()
+    vm.start()
+
+    snapshot = vm.snapshot_full()
+
+    snap_editor = host.get_binary("snapshot-editor")
+
+    # Test that MIDR_EL1 is in the snapshot
+    cmd = [
+        str(snap_editor),
+        "info-vmstate",
+        "vcpu-states",
+        "--vmstate-path",
+        str(snapshot.vmstate),
+    ]
+    _, stdout, _ = utils.run_cmd(cmd)
+    assert stdout.find(str(MIDR_EL1)) != -1
+
+    # Remove MIDR_EL1 register from the snapshot
+    cmd = [
+        str(snap_editor),
+        "edit-vmstate",
+        "remove-regs",
+        "--vmstate-path",
+        str(snapshot.vmstate),
+        "--output-path",
+        str(snapshot.vmstate),
+        str(MIDR_EL1),
+    ]
+    utils.run_cmd(cmd)
+
+    # Test that MIDR_EL1 is not in the snapshot
+    cmd = [
+        str(snap_editor),
+        "info-vmstate",
+        "vcpu-states",
+        "--vmstate-path",
+        str(snapshot.vmstate),
+    ]
+    _, stdout, _ = utils.run_cmd(cmd)
+    assert stdout.find(str(MIDR_EL1)) == -1
```

Comment:
I think this test can be more generic. I can think of at least two cases to test here:

1. We should check that no other registers were touched or added.
2. We should check that if we ask the tool to remove a non-existent register it does the right thing.